### PR TITLE
London Class_8 - Busra_Erdogan - HTML-CSS-Coursework-Week2

### PR DESCRIPTION
**Pull request by BusraElmasErdogan**
HTML_CSS Works for Karma Webpage

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Busra Erdogan
- Your City: London
- Your Slack Name: BusraErdogan

# Homework Details

- Module: HTML-CSS Coursework
- Week: 2

# Notes

- What did you find easy? 

- What did you find hard?
- After forking the project, I started working without creating a new branch. I noticed this before PR  and it was hard to find the update the master feature in branch menu on Github Desktop.

- What do you still not understand? 

- Any other notes?

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,19 +1,91 @@
...
-</body>
+      <nav>
+        <ul class="nav-list">
+          <li class="nav-item-meet-karma">
+            <a class="nav-link-meet-karma" href="#">Meet Karma</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="#">How it Works</a>
+          </li>
+          <li class="nav-item">
```
Comment (kolomiets) [inline index.html:32]: It looks we repeat nav-item and nav-link styles a lot. Let's drop this code duplication from HTML and handle the styling in CSS

File: index.html [hunk 2]
```diff
@@ -1,19 +1,91 @@
...
+        <div class="karma">
+          <h1>Introducing Karma</h1>
+          <p>Bring WiFi with you, everywhere you go.</p>
+          <button class="button">Learn More</button>
+        </div>
+      </section>
+
+      <section class="details">
+        <h1>Everyone needs a little Karma.</h1>
+        <div class="article-container">
```
Comment (kolomiets) [inline index.html:59]: Articles are very close to each other - check how they look like on the target design (png image)

File: index.html [hunk 3]
```diff
@@ -1,19 +1,91 @@
...
+          <article>
+            <img src="img/icon-refill.svg" alt="" class="" />
+            <p>Pay as You Go</p>
+          </article>
+        </div>
+      </section>
+    </main>
+
+    <footer>
+      <div class="line"></div>
```
Comment (kolomiets) [inline index.html:77]: Let's use <hr/> here - this is what it is, right?

File: index.html [hunk 4]
```diff
@@ -1,19 +1,91 @@
...
+      <h5>Join us on</h5>
```
Comment (kolomiets) [inline index.html:78]: Why this is `h5`? You can use `h1` for the purposes of the main header in different parts of the page and style them differently in CSS. `h1` in header may look very different from `h1` in footer. 

`h5` kind of assumes that there are other header levels but it is not the case here.

**Comment by kolomiets** [comment]
Very good job - clean, very easy to follow the page structure, style names are simple and in "logical" order - it's a pleasure to read! 

There are some problems in the mobile mode - the page falls apart. I assume we haven't covered this material yet. Other than that - very-very good! I added just a couple of minor comments/questions.

File: index.html (hunk 2 above)
Comment (BusraElmasErdogan) [reply index.html:59]: I updated article container paddings 29 to 23rem.

File: index.html (hunk 3 above)
Comment (BusraElmasErdogan) [reply index.html:77]: Yes, definitely.

File: index.html (hunk 4 above)
Comment (BusraElmasErdogan) [reply index.html:78]: Ok got it. Done.

File: index.html (hunk 1 above)
Comment (BusraElmasErdogan) [reply index.html:32]: I don't think I understand what you mean. I had already styled in CSS not HTML. These are class names. Do you want me to remove class names(nav-item, nav-link) ?

Comment (kolomiets) [reply index.html:32]: Currently you have:

```html
<ul class="nav-list">
  <li class="nav-item">
    <a class="nav-link" href="#">Blog</a>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="#">How it Works</a>
  </li>
</ul>
```

What I'm suggesting is:

```html
<ul>
  <li>
    <a href="#">Blog</a>
  </li>
  <li>
    <a href="#">How it Works</a>
  </li>
</ul>
```

You can still apply the styles you want in CSS. But this makes HTML more compact and readable. 

Usually, when you repeat the same thing over and over (styles for each navigation item in our case) - it's a good sign there is a way to simplify and reduce the code duplication.

File: index.html (hunk 3 above)
Comment (kolomiets) [reply index.html:77]: What I meant is the following HTML:

```html
<footer>
  <hr/>
  <!-- the rest of the footer code -->
</footer>
```

Using a div to draw a horizontal line feels strange as we have a semantic element designed for this very purpose - `hr`. Definitely an optional task - try to replace `div` with `hr` and see if it works. Can you make it look the same as before?

**Pull request closed without merging**